dev/com.ibm.ws.webcontainer.servlet.4.0_fat/fat/src/com/ibm/ws/fat/wc/tests/WC5JakartaServletTest.java

```diff
@@ -102,35 +103,94 @@ public void test_SuppressHtmlRecursiveErrorOutput() throws Exception {
     }
 
     /**
-     * Common request helper >
      *
-     * scheme - http or https (empty will default to http)
-     * httpMethod - GET/POST/PUT .. (default is GET)
-     * uri - /ContextRoot/path?queryName=queryValue
-     * statusCode
-     * expectedResponse
-     * notExpectedResponse
+     * Ensure query param with no equals is registered as an empty string for servlet 5.0.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testNoEqualsQueryParameter() throws Exception {
+        requestHelper("http", "GET", "/" + APP_NAME + "/query?test", "", "SUCCESS!", "");
+    }
+
+    /*
+     * Test enablePostOnlyJSecurityCheck which is default to true for Servlet 5.0+
+     * Any method, but POST, will get a 404 status
+     *
+     * POST test to verify if the authentication process can redirect to ReLogin.jsp for form login; no login is needed.
+     */
+    @Test
+    public void test_POST_j_security_check() throws Exception {
+        requestHelper("http", "POST", "/" + APP_NAME + "/j_security_check", "200", "ReLogin", "");
+    }
+
+    /*
+     * Default enablePostOnlyJSecurityCheck not allow GET /j_security_check. Response with a 404
+     */
+    @Test
+    public void test_Default_j_security_check() throws Exception {
+        requestHelper("http", "GET", "/" + APP_NAME + "/j_security_check", "404", "", "");
+    }
+
+    /*
+     * Test enablePostOnlyJSecurityCheck = "false" to allow GET /j_security_check. Redirect to ReLogin.jsp; no login is needed
+     *
+     * Load EnablePostOnlyJSecurityCheckServer.xml which contains custom property
+     */
+    @Test
+    public void test_GET_j_security_check() throws Exception {
+        server.saveServerConfiguration();
+        ServerConfiguration configuration = server.getServerConfiguration();
+        LOG.info("Server configuration that was saved: " + configuration);
+        server.setMarkToEndOfLog();
+        server.setServerConfigurationFile("serverConfigs/EnablePostOnlyJSecurityCheckServer.xml");
+
+        //Wait for "CWWKG0017I: The server configuration was successfully updated in"
+        server.waitForStringInLogUsingMark("CWWKG0017I:.*");
```

Comment:
I think using `waitForConfigUpdateInLogUsingMark` would work better you can also ensure the app is up as well. You can look at other uses in this bucket for some examples.

Reply:
You may need to wait for the feature update to complete as well `waitForFeatureUpdateCompleted` parameter to the above mentioned method.

Reply:
Using:
      `  server.waitForConfigUpdateInLogUsingMark(Collections.singleton(APP_NAME), true, "CWWKG0017I:.*successfully.*");`

but I keep getting search not found and timeout

`2021-09-09-11:15:32:297 Timed out waiting for [] and/or searching for [CWWKG001[7-8]I, CWWKF0008I:] in log file: /Users/pmdinh/1_WAS/ASLibertyGit/Jan16_21/open-liberty/dev/build.image/wlp/usr/servers/servlet50_wcServer/logs/messages.log. Extra info: value of (watchFor.size() > 0): true, value of (notStartedAppNames.size() > 0): false
`

even though the message show up in both trace and message.log

`[9/9/21, 11:15:19:991 EDT] 00000036 ConfigRefresh A   CWWKG0017I: The server configuration was successfully updated in 0.024 seconds.`

Also, the app is not restarted when changing the server configuration.

I'm going with the `waitForStringInLogUsingMark` for now.